pdf_parser/pdf_parse.py

```diff
@@ -29,8 +30,9 @@ def parse_pdf_document(document):
         with open(os.devnull, 'w') as FNULL:
             subprocess.check_call(cmd, stdout=FNULL, stderr=FNULL)
 
-        with open(parsed_path, 'rb') as html_file:
-            soup = bs(html_file.read(), 'html.parser')
+        # Try to get file stats in order to check both its existance
+        # and if it has some content
+        os.stat(parsed_path)
```

Comment:
It's probably best two have two separate try/except blocks

```python
try:
    with open(os.devnull, 'w') as FNULL:
        subprocess.check_call(cmd, stdout=FNULL, stderr=FNULL)
except subprocess.CalledProcessError as e:
    logger.warning(...)
    return None, None

try:
    st = os.stat(parsed_path)
except OSError as e:
    if e.errno != errno.ENOENT:
        raise
    logger.warning(...)
    return None, None

if st.st_size == 0:
    logger.warning(...) # empty file
    return None, None

with open(parsed_path, 'rb') as html_file:
    ...
```

But, maybe I'm missing something...

Reply:
I though it was best to put it on the same try block, but if you see no objection to it, I find it clearer in a separate block. Will update